cmd/server/pactasrv/pactasrv.go

```diff
@@ -211,23 +215,37 @@ type actorInfo struct {
 	IsSuperAdmin bool
 }
 
-func (s *Server) getActorInfoOrFail(ctx context.Context) (*actorInfo, error) {
+var anonymousActorInfo = actorInfo{}
+
+func (s *Server) getActorInfoOrFail(ctx context.Context) (actorInfo, error) {
 	actorUserID, err := getUserID(ctx)
 	if err != nil {
-		return nil, err
+		return anonymousActorInfo, err
 	}
 	actorOwnerID, err := s.getUserOwnerID(ctx)
 	if err != nil {
-		return nil, err
+		return anonymousActorInfo, err
 	}
 	actorIsAdmin, actorIsSuperAdmin, err := s.isAdminOrSuperAdmin(ctx)
 	if err != nil {
-		return nil, err
+		return anonymousActorInfo, err
 	}
-	return &actorInfo{
+	return actorInfo{
 		UserID:       actorUserID,
 		OwnerID:      actorOwnerID,
 		IsAdmin:      actorIsAdmin,
 		IsSuperAdmin: actorIsSuperAdmin,
 	}, nil
 }
+
+func (s *Server) getActorInfoOrAnon(ctx context.Context) (actorInfo, error) {
+	_, err := getUserID(ctx)
+	if err != nil {
```

Comment:
_technically_ there are other reasons we can fail to get the user ID from the context, but calling the user 'anonymous' regardless is probably fine

Reply:
Yeah it's definitely an edge case - I doubt we'll hit it more than a handful of times over the operation of the site - but if/when user lookup fails, that really should be handled differently than an anonymous user to the site, which is why this differentiation feels important. It's the `(*x, err)` pattern of go after all!